components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts

```diff
@@ -389,9 +389,16 @@ export class ReportingComponent implements OnInit, OnDestroy {
     const queryParams = {...this.route.snapshot.queryParams};
 
     const filteredParams = pickBy((_value, key) => {
+
+      // Only handles control_tag filters
+      if (key.includes('control_tag')) {
+        return false;
+      }
+
```

Comment:
While this solves the immediate problem in what I feel is a pragmatic way, I don't believe in its value long term.  It also begins to expose future issues.  Currently, in order to remove the applied query filters, we are looping through an array that contains all of our _first_ level filter keys, checking it agains the query parameters, and then removing if it is a match.  This does not and will not account for any second/third level filtering we choose to do in the future, because with second level filtering, (i.e. control_tag) we end up with a key that looks like `control_tag:something`.  I'm not sure we ever want that many keys in the array we're checking against.

Reply:
a possible option might be to split the key at `:` and return the first element of that array.  If i'm thinking correctly, it will return the expected key even if a `:` doesn't exist.